
Regretably, with GCC 4.4 or later this is the only flag combination
I could find that doesn't cause kernel panics CPU triple-faults or
other kind of severe breakage.

Index: kfreebsd-10/sys/conf/kern.pre.mk
===================================================================
--- kfreebsd-10.orig/sys/conf/kern.pre.mk	2012-11-02 12:57:20.000000000 -0700
+++ kfreebsd-10/sys/conf/kern.pre.mk	2012-11-02 12:57:46.879211260 -0700
@@ -47,6 +47,9 @@
 .if !empty(COPTFLAGS:M-O[23s]) && empty(COPTFLAGS:M-fno-strict-aliasing)
 COPTFLAGS+= -fno-strict-aliasing
 .endif
+.if !empty(COPTFLAGS:M-O[23s])
+COPTFLAGS+= -O1
+.endif
 .if !defined(NO_CPU_COPTFLAGS)
 COPTFLAGS+= ${_CPUCFLAGS}
 .endif
Index: kfreebsd-10/sys/conf/kmod.mk
===================================================================
--- kfreebsd-10.orig/sys/conf/kmod.mk	2012-11-02 12:57:46.469212125 -0700
+++ kfreebsd-10/sys/conf/kmod.mk	2012-11-02 12:57:46.879211260 -0700
@@ -87,6 +87,9 @@
 .if !empty(CFLAGS:M-O[23s]) && empty(CFLAGS:M-fno-strict-aliasing)
 CFLAGS+=	-fno-strict-aliasing
 .endif
+.if !empty(CFLAGS:M-O[23s])
+CFLAGS+=	-O1
+.endif
 WERROR?=	-Werror
 CFLAGS+=	${WERROR}
 CFLAGS+=	-D_KERNEL
